taxonomy-editor-frontend/src/components/useFetch.jsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from "react";
+
+const useFetch = (url) => {    
+    const [data, setData] = useState(null)
+    const [isPending, setIsPending] = useState(true)
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        const abortCont = new AbortController();
+        fetch(url, {signal: abortCont.signal})
+            .then(res => { 
+                if (!res.ok) {
+                    throw Error("Could not fetch the data for resource!")
+                }
+                return res.json();
+            })
+            .then(data => {
+                setData(data);
+                setIsPending(false);
+                setError(null);
+            })
+            .catch(err => {
+                if (err.name === 'AbortError') {
+                    console.log("Fetch aborted!");
+                } else {
+                    setIsPending(false);
+                    console.log(err);

```

Comment:
logs are good during development, but in general not really useful to deliver them to production